cmd/kubelet/app/server.go

```diff
@@ -122,182 +105,63 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
-		PodInfraContainerImage:       s.PodInfraContainerImage,
-		Port:                           uint(s.Port),
-		ReadOnlyPort:                   uint(s.ReadOnlyPort),
-		RegisterNode:                   s.RegisterNode,
-		RegisterSchedulable:            s.RegisterSchedulable,
-		RegistryBurst:                  int(s.RegistryBurst),
-		RegistryPullQPS:                float64(s.RegistryPullQPS),
-		ResolverConfig:                 s.ResolverConfig,
-		Reservation:                    *reservation,
-		KubeletCgroups:                 s.KubeletCgroups,
-		RktPath:                        s.RktPath,
-		RktAPIEndpoint:                 s.RktAPIEndpoint,
-		RktStage1Image:                 s.RktStage1Image,
-		RootDirectory:                  s.RootDirectory,
-		SeccompProfileRoot:             s.SeccompProfileRoot,
-		Runonce:                        s.RunOnce,
-		SerializeImagePulls:            s.SerializeImagePulls,
-		StandaloneMode:                 (len(s.APIServerList) == 0),
-		StreamingConnectionIdleTimeout: s.StreamingConnectionIdleTimeout.Duration,
-		SyncFrequency:                  s.SyncFrequency.Duration,
-		SystemCgroups:                  s.SystemCgroups,
-		TLSOptions:                     tlsOptions,
-		Writer:                         writer,
-		VolumePlugins:                  ProbeVolumePlugins(s.VolumePluginDir),
-		OutOfDiskTransitionFrequency:   s.OutOfDiskTransitionFrequency.Duration,
-		HairpinMode:                    s.HairpinMode,
-		BabysitDaemons:                 s.BabysitDaemons,
-		ExperimentalFlannelOverlay:     s.ExperimentalFlannelOverlay,
-		NodeIP:         net.ParseIP(s.NodeIP),
-		EvictionConfig: evictionConfig,
-		PodsPerCore:    int(s.PodsPerCore),
+	return &kubelet.KubeletConfig{
+		Auth:              nil, // default does not enforce auth[nz]
+		CAdvisorInterface: nil, // launches background processes, not set here
+		Cloud:             nil, // cloud provider might start background processes
+		ContainerManager:  nil,
+		DockerClient:      dockertools.ConnectToDockerOrDie(s.DockerEndpoint, s.RuntimeRequestTimeout.Duration), // TODO(random-liu): Set RuntimeRequestTimeout for rkt.
+		KubeClient:        nil,
+		Mounter:           mounter,
+		NetworkPlugins:    ProbeNetworkPlugins(s.NetworkPluginDir),
+		OOMAdjuster:       oom.NewOOMAdjuster(),
+		OSInterface:       kubecontainer.RealOS{},
+		TLSOptions:        tlsOptions,
+		Writer:            writer,
+		VolumePlugins:     ProbeVolumePlugins(s.VolumePluginDir),
 	}, nil
 }
 
+// TODO(mtaufen): Once KubeletConfig is gone, how will we allow people to pass
+//                arbitrary config. I've transferred a lot of the behavior of
+//                UnsecuredKubeletConfig to NewMainKubelet, which should be fine
+//                for the most part, because mesos relies on that for the foundation
+//                of their config. However, there is the issue that they use
+//                kcfg != nil to avoid initializing the cloud provider. We could
```

Comment:
avoiding init of the mesos cloud provider is very important: we definitely don't want the mesos cloud provider spinning up on every node.